### PR TITLE
test: move provisioner-cli to test folder

### PR DESCRIPTION
**Pull request by huoqifeng**
Fixes: #731

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -90,6 +90,10 @@ else
 	$(error CLOUD_PROVIDER is not set)
 endif
 
+.PHONY: provisioner-cli
```
Comment (bpradipt) [inline Makefile:93]: I think having a separate Makefile inside `test/tools` will be more apt and give flexibility to add targets fpr new tools as well as any tests for the tools itself.

Comment (huoqifeng) [reply Makefile:93]: > I think having a separate Makefile inside `test/tools` will be more apt and give flexibility to add targets fpr new tools as well as any tests for the tools itself.

Thanks @bpradipt sounds also better to move readme to tools root directory.

Comment (bpradipt) [reply Makefile:93]: Right, both README and a separate Makefile in `tools` root directory makes sense to me.
Thanks @huoqifeng

File: test/tools/Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# (C) Copyright Confidential Containers Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+.PHONY: clean provisioner-cli
+
+ARCH        ?= $(subst x86_64,amd64,$(shell uname -m))
+GOOPTIONS   ?= GOOS=linux GOARCH=$(ARCH) CGO_ENABLED=0
+GOFLAGS     ?= -tags=$(aws azure ibmcloud vsphere libvirt)
```
Comment (bpradipt) [inline test/tools/Makefile:8]: should this be only for azure ibmcloud and libvirt for now ?

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @huoqifeng 
Asked a question but it's not a blocker

**Comment by huoqifeng** [comment]
> /lgtm Thanks @huoqifeng Asked a question but it's not a blocker

Thanks @bpradipt , yes, we supports these providers only so far, fixed it.

**Review by liudalibj — APPROVED** [review]
LGTM

**Comment by stevenhorsman** [comment]
@huoqifeng - I appreciate I'm a bit late here, but I have a question:
- Previously I could run `make BUILTIN_CLOUD_PROVIDERS="ibmcloud" build` to just run the build of the ibmcloud provisioning part, but the `BUILTIN_CLOUD_PROVIDERS` option isn't used in the new `makefile` now? This means that I now need to install libvirt set-up to provision IBM Cloud, or I get failures:
```
/go/src/github.com/huoqifeng/cloud-api-adaptor/test/tools# make BUILTIN_CLOUD_PROVIDERS="ibmcloud" provisioner-cli
GOOS=linux GOARCH=amd64 CGO_ENABLED=1 go build "-tags=azure,ibmcloud,libvirt" -o caa-provisioner-cli "provisioner-cli/main.go"
# pkg-config --cflags  -- libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt
Package libvirt was not found in the pkg-config search path.
Perhaps you should add the directory containing `libvirt.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libvirt' found
Package libvirt was not found in the pkg-config search path.
Perhaps you should add the directory containing `libvirt.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libvirt' found
x 25
```
As we add support for more cloud providers this might get more annoying, so I wondered if we could re-instate the equivalent function?

File: test/tools/provisioner-cli/main.go [hunk 3]
```diff
@@ -32,12 +32,12 @@ func initLogger() {
...
 
 // export LOG_LEVEL="trace|debug"
-// export CLOUD_PROVIDER="ibmcloud"
+// export CLOUD_PROVIDER="ibmcloud|azure|libvirt"
 // export TEST_E2E_PROVISION_FILE="/root/provision_ibmcloud.properties"
 // export TEST_E2E_PODVM_IMAGE="/root/e2e-test-image-amd64-20230308.qcow2
 // export TEST_E2E_PROVISION="yes"
 // cd test/e2e
-// ../../cluster-provisioner -action=provision | deprovision | uploadimage
+// ./cluster-provisioner -action=provision | deprovision | uploadimage
```
Comment (stevenhorsman) [inline test/tools/provisioner-cli/main.go:40]: I think this has changed to `./caa-provisioner-cli` now?

Comment (huoqifeng) [reply test/tools/provisioner-cli/main.go:40]: Yes, I'll fix it, thank you @stevenhorsman

**Pull request merged**